### DISCUSSION TITLE
Giving up on next-auth or is there a solution that actually works?

### DISCUSSION THREAD

**Discussion by acdpdevs** [nextauthjs/next-auth#9629] [Help]
So, my company used next auth as of last year, only with the oAuth providers (google and facebook). Now we want to use this tool for some of our other products that we develop for b2b and as such we need to use the credentials provider, since no b2b client will ever use a google account to sign in into a product they use internally. The problem that we face is that we can't do anything with the credentials provider. We need to save the session into the database, such as if an admin or someone wants to kill somebody session with a click of a button from a dashboard, they can. 

The credentials provides is so limited and so useless, we wonder why the next auth added it in the first place. :( 
Did anyone have find a way of making the database session strategy working with the credentials provider? We tried some of the solutions we found, yet none of them seems to work? 

Thank you?

**Comment by ivanjeremic** [comment]
1: There are thousands of b2b products that companies use and sign up with OAuth.
2: I would implement auth by myself if I where you.

**Comment by djflorio** [comment]
Look into [Lucia](https://lucia-auth.com/). I'm using it for a Next.js project with Postgres and happy with it so far. You pretty much have full control over everything because you add a user table, key table (for storing hashed passwords), and session table.

**Reply by ivanjeremic** [reply]
That looks interesting to me because it supports "unstorage" which I heavily use for my own db.

**Reply by ivanjeremic** [reply]
Did you find Lucia just now or did you use it before? If you used it before why did you even start using next-auth?

**Reply by djflorio** [reply]
I didn't know about it before I started using next-auth. I'm currently working on 2 different next.js sites. The first one I started working on a year ago and uses next-auth for Azure OAuth. I started work on the 2nd site about a month ago, and I needed email/password authentication rather than OAuth. Since next-auth actively discourages email/password authentication, I had to find something else, which is when I found Lucia.

**Reply by acdpdevs** [reply]
Does lucia works with MongoDb?

**Reply by djflorio** [reply]
@acdpdevs it looks like it has support for Mongoose. I've only tried it with Postgres.
